src/main.rs

```diff
@@ -44,6 +45,8 @@ extern "C" {
 #[allow(non_upper_case_globals)]
 pub(crate) static _stack_start: u8 = 0;
 
+static BENCH: Mutex<Benchmark> = Mutex::new(Benchmark::new());
```

Comment:
Another important comment, but we can handle that later when we move to multi-core: this would need to be thread local, not a shared global. Otherwise multiple cores would increment the same counters, which might not be what we want.

Setting-up thread local variables is a bit complicated though, so we can delay that for now until we need to benchmark on multicore machines.

Reply:
Yes I was thinking about it and had the same conclusion.